### PR TITLE
Initial commit for the Websites commandlets for ARM

### PR DESCRIPTION
**Pull request by NaveenGoli**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@NaveenGoli__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (ngoli). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: src/AzurePowershell.sln [hunk 1]
```diff
@@ -161,7 +161,7 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Commands.StreamAnalytics.Te
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Commands.Insights", "ResourceManager\Insights\Commands.Insights\Commands.Insights.csproj", "{DEA446A1-84E2-46CC-B780-EB4AFDE2460E}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Commands.Insights.Test", "ResourceManager\Insights\Commands.Insights.Test\Commands.Insights.Test.csproj", "{469F20E0-9D40-41AD-94C3-B47AD15A4C00}"
```
Comment (amitapl) [inline src/AzurePowershell.sln]: why did you remove this project?

File: src/ResourceManager/Websites/Commands.Websites/AzureResourceManager.psd1 [hunk 2]
```diff
@@ -0,0 +1,107 @@
+#
```
Comment (amitapl) [inline src/ResourceManager/Websites/Commands.Websites/AzureResourceManager.psd1]: is this file required per resource?

File: src/ResourceManager/Websites/Commands.Websites/Cmdlets/NewAzureWebsites.cs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                if (_websitesClient == null)
+                {
+                    _websitesClient = new WebsitesClient(CurrentContext);
+                }
+                return _websitesClient;
+            }
+        }
+
+
+
```
Comment (amitapl) [inline src/ResourceManager/Websites/Commands.Websites/Cmdlets/NewAzureWebsites.cs]: extra \n

File: src/ResourceManager/Websites/Commands.Websites/Cmdlets/NewAzureWebsites.cs [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        [Parameter(Position = 2, Mandatory = false, HelpMessage = "The name of the website slot.")]
+        [ValidateNotNullOrEmptyAttribute]
+        public string SlotName { get; set; }
+
+        [Parameter(Position = 3, Mandatory = true, HelpMessage = "The Location of the Website eg: West US.")]
+        public string Location { get; set; }
+
+        [Parameter(Position = 4, Mandatory = true, HelpMessage = "The name of the web hosting plan eg: Default1.")]
+        public string WebHostingPlan { get; set; }
```
Comment (amitapl) [inline src/ResourceManager/Websites/Commands.Websites/Cmdlets/NewAzureWebsites.cs]: what happens when the web hosting plan doesnt exist?

File: src/ResourceManager/Websites/Commands.Websites/Cmdlets/RemoveAzureWebsite.cs [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+            set { this.deleteEmptyServerFarmBydefault = value; }
+        }
+
+        [Parameter(Position = 5, Mandatory = true, HelpMessage = "Delete website metrics?")]
+        [ValidateNotNullOrEmptyAttribute]
+        public bool DeleteMetrics 
+        { 
+            get { return this.deleteMetricsBydefault; }
+            set { this.deleteMetricsBydefault = value; }
+        }
```
Comment (amitapl) [inline src/ResourceManager/Websites/Commands.Websites/Cmdlets/RemoveAzureWebsite.cs]: Do we want default values for these booleans?

File: src/ResourceManager/Websites/Commands.Websites/Cmdlets/RemoveAzureWebsite.cs [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+namespace Microsoft.Azure.Commands.Websites.Cmdlets
+{
+    /// <summary>
+    /// this commandlet will let you create a new backuppolicy with schedules
+    /// </summary>
+    [Cmdlet(VerbsCommon.Remove, "AzureWebsite")]
+    public class RemoveAzureWebsiteCmdlet : AzurePSCmdlet
+    {
+
+        private bool deleteSlotsBydefault = false;
```
Comment (amitapl) [inline src/ResourceManager/Websites/Commands.Websites/Cmdlets/RemoveAzureWebsite.cs]: `deleteSlotsBydefault` --> `deleteSlotsByDefault`

File: src/ResourceManager/Websites/Commands.Websites/Cmdlets/NewAzureWebsites.cs [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+using System.Net;
+using Microsoft.Azure;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using Microsoft.Azure.Commands.Websites.Utilities;
+
+
+namespace Microsoft.Azure.Commands.Websites.Cmdlets
+{
+    /// <summary>
+    /// this commandlet will let you create a new backuppolicy with schedules
```
Comment (galcho) [inline src/ResourceManager/Websites/Commands.Websites/Cmdlets/NewAzureWebsites.cs]: wrong comment - it is same for the files below too

File: src/ResourceManager/Websites/Commands.Websites/Cmdlets/RemoveAzureWebsite.cs [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+
+        [Parameter(Position = 4, Mandatory = true, HelpMessage = "Delete empty website server farm?")]
+        [ValidateNotNullOrEmptyAttribute]
+        public bool DeleteEmptyServerFarm
+        {
+            get { return this.deleteEmptyServerFarmBydefault; }
+            set { this.deleteEmptyServerFarmBydefault = value; }
+        }
+
+        [Parameter(Position = 5, Mandatory = true, HelpMessage = "Delete website metrics?")]
```
Comment (galcho) [inline src/ResourceManager/Websites/Commands.Websites/Cmdlets/RemoveAzureWebsite.cs]: are those mandatory?

File: src/ResourceManager/Websites/Commands.Websites/Cmdlets/StopAzureWebsite.cs [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+{
+    /// <summary>
+    /// this commandlet will let you create a new backuppolicy with schedules
+    /// </summary>
+    [Cmdlet(VerbsLifecycle.Stop, "AzureWebsite")]
+    public class StopAzureWebsiteCmdlet : AzurePSCmdlet
+    {
+        [Parameter(Position = 0, Mandatory = true, HelpMessage = "The name of the resource group.")]
+        [ValidateNotNullOrEmptyAttribute]
+        public string ResourceGroupName { get; set; }
```
Comment (amitapl) [inline src/ResourceManager/Websites/Commands.Websites/Cmdlets/StopAzureWebsite.cs]: These 3 parameters: `ResourceGroupName`. `WebsiteName` and `SlotName` are repeated a couple of times, maybe its worth creating a base `WebsiteCmdlet` class.

File: src/ResourceManager/Websites/Commands.Websites/Commands.Websites.csproj [hunk 10]
```diff
@@ -0,0 +1,218 @@
...
+    </ProjectReference>
+    <ProjectReference Include="..\..\Batch\Commands.Batch\Commands.Batch.csproj">
+      <Project>{d470e50a-9607-48d6-a924-4f9f86502704}</Project>
+      <Name>Commands.Batch</Name>
+    </ProjectReference>
+    <ProjectReference Include="..\..\KeyVault\Commands.KeyVault\Commands.KeyVault.csproj">
+      <Project>{9ffc40cc-a341-4d0c-a25d-dc6b78ef6c94}</Project>
+      <Name>Commands.KeyVault</Name>
+    </ProjectReference>
+    <ProjectReference Include="..\..\RedisCache\Commands.RedisCache\Commands.RedisCache.csproj">
```
Comment (amitapl) [inline src/ResourceManager/Websites/Commands.Websites/Commands.Websites.csproj]: I think redis, storsimple and similar are not really required for this project.

File: src/ResourceManager/Websites/Commands.Websites/Cmdlets/StopAzureWebsite.cs (hunk 9 above)
Comment (galcho) [inline src/ResourceManager/Websites/Commands.Websites/Cmdlets/StopAzureWebsite.cs]: and private WebsitesClient WebsitesClient as well

File: src/ResourceManager/Websites/Commands.Websites/Utilities/WebsitesClient.cs [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+{
+
+    public class WebsitesClient
+    {
+        public WebsitesClient(AzureContext context)
+        {
+            this.Resourcesclient = AzureSession.ClientFactory.CreateClient<ResourceManagementClient>(context, AzureEnvironment.Endpoint.ResourceManager);
+            this.WrappedWebsitesClient = AzureSession.ClientFactory.CreateClient<WebSiteManagementClient>(context, AzureEnvironment.Endpoint.ResourceManager);
+
+        }
```
Comment (galcho) [inline src/ResourceManager/Websites/Commands.Websites/Utilities/WebsitesClient.cs]: new line for readability

**Comment by amitapl** [comment]
Is the file `src/ResourceManager/Websites/Commands.Websites/MSSharedLibKey.snk` required in the repo?

File: src/ResourceManager/Websites/Commands.Websites/Utilities/WebsitesClient.cs [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+        public System.Net.HttpStatusCode RemoveWebsite(string resourceGroupName, string webSiteName, string slotName, bool deleteEmptyServerFarmBydefault, bool deleteMetricsBydefault, bool deleteSlotsBydefault)
+        {
+            WebSiteDeleteParameters webSiteDelParams = new WebSiteDeleteParameters(deleteEmptyServerFarmBydefault, deleteMetricsBydefault, deleteSlotsBydefault);
+            
+            var startedWebsite = WrappedWebsitesClient.WebSites.Delete(resourceGroupName, webSiteName, slotName, webSiteDelParams);
+            return startedWebsite.StatusCode;
+        }
+
+   
+
```
Comment (galcho) [inline src/ResourceManager/Websites/Commands.Websites/Utilities/WebsitesClient.cs]: extra new lines

**Comment by ogail** [comment]
@azuresdkci add to whitelist

File: src/ResourceManager/Websites/Commands.Websites/Cmdlets/NewAzureWebsites.cs (hunk 4 above)
Comment (NaveenGoli) [inline src/ResourceManager/Websites/Commands.Websites/Cmdlets/NewAzureWebsites.cs]: Whatever error I get from API that is returned.

**Comment by NaveenGoli** [comment]
I have incorporated all the feedback changes. please review..

File: src/ResourceManager/Websites/Commands.Websites/Cmdlets/NewAzureWebsites.cs [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+using Microsoft.Azure.Management.WebSites;
+using System.Net.Http;
+using System.Threading;
+using Microsoft.IdentityModel.Clients.ActiveDirectory;
+using System.Net;
+using Microsoft.Azure;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using Microsoft.Azure.Commands.Websites.Utilities;
+
+
```
Comment (amitapl) [inline src/ResourceManager/Websites/Commands.Websites/Cmdlets/NewAzureWebsites.cs]: you have some extra \n throughout

File: src/ResourceManager/Websites/Commands.Websites/Cmdlets/RemoveAzureWebsite.cs [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+{
+    /// <summary>
+    /// this commandlet will let you delete an Azure website
+    /// </summary>
+    [Cmdlet(VerbsCommon.Remove, "AzureWebsite")]
+    public class RemoveAzureWebsiteCmdlet : AzurePSCmdlet
+    {
+
+       //always delete the slots, 
+        private bool deleteSlotsBydefault = true;
```
Comment (amitapl) [inline src/ResourceManager/Websites/Commands.Websites/Cmdlets/RemoveAzureWebsite.cs]: deleteSlotsBy**d**efault --> deleteSlotsBy**D**efault
Although I think these 3 member are not required at all.

File: src/ResourceManager/Websites/Commands.Websites/Cmdlets/RemoveAzureWebsite.cs [hunk 15]
```diff
@@ -0,0 +1,99 @@
...
+        [Parameter(Position = 0, Mandatory = true, HelpMessage = "The name of the resource group.")]
+        [ValidateNotNullOrEmptyAttribute]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Position = 1, Mandatory = true, HelpMessage = "The name of the website.")]
+        [ValidateNotNullOrEmptyAttribute]
+        public string WebsiteName { get; set; }
+
+
+        [Parameter(Mandatory = false, HelpMessage = "Force to delete website if it has slots.")]
```
Comment (amitapl) [inline src/ResourceManager/Websites/Commands.Websites/Cmdlets/RemoveAzureWebsite.cs]: HelpMessage should be updated (not related to slots)

File: src/ResourceManager/Websites/Commands.Websites/Cmdlets/StopAzureWebsite.cs (hunk 9 above)
Comment (amitapl) [inline src/ResourceManager/Websites/Commands.Websites/Cmdlets/StopAzureWebsite.cs]: Reminder to create a base AzureWebsiteCmdlet class

File: src/ResourceManager/Websites/Commands.Websites/Commands.Websites.csproj [hunk 16]
```diff
@@ -0,0 +1,179 @@
...
+      <Project>{4900ec4e-8deb-4412-9108-0bc52f81d457}</Project>
+      <Name>Commands.Utilities</Name>
+    </ProjectReference>
+  </ItemGroup>
+  <ItemGroup>
+    <Content Include="Microsoft.Azure.Commands.Websites.dll-Help.xml">
+      <SubType>Designer</SubType>
+      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+    </Content>
+    <None Include="MSSharedLibKey.snk" />
```
Comment (amitapl) [inline src/ResourceManager/Websites/Commands.Websites/Commands.Websites.csproj]: is this file required?

File: src/ResourceManager/Websites/Commands.Websites/Properties/AssemblyInfo.cs [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+[assembly: AssemblyTitle("Microsoft Azure Powershell - Resource Manager")]
```
Comment (amitapl) [inline src/ResourceManager/Websites/Commands.Websites/Properties/AssemblyInfo.cs]: "Azure Websites"

File: src/ResourceManager/Websites/Commands.Websites/Properties/Resources.resx [hunk 18]
```diff
@@ -0,0 +1,228 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (amitapl) [inline src/ResourceManager/Websites/Commands.Websites/Properties/Resources.resx]: Are you using any resource from this new file?

File: src/ResourceManager/Websites/Commands.Websites/Utilities/WebsitesClient.cs [hunk 19]
```diff
@@ -0,0 +1,97 @@
...
+            this.WrappedWebsitesClient = AzureSession.ClientFactory.CreateClient<WebSiteManagementClient>(context, AzureEnvironment.Endpoint.ResourceManager);
+
+        }
+        public WebSiteManagementClient WrappedWebsitesClient
+        {
+            get;
+            private set;
+        }
+
+        public ResourceManagementClient Resourcesclient
```
Comment (amitapl) [inline src/ResourceManager/Websites/Commands.Websites/Utilities/WebsitesClient.cs]: property is not used

File: src/ResourceManager/Websites/Commands.Websites/Commands.Websites.csproj (hunk 16 above)
Comment (NaveenGoli) [inline src/ResourceManager/Websites/Commands.Websites/Commands.Websites.csproj]: you mean ..snk - Yes

File: src/ResourceManager/Websites/Commands.Websites/Properties/Resources.resx (hunk 18 above)
Comment (NaveenGoli) [inline src/ResourceManager/Websites/Commands.Websites/Properties/Resources.resx]: Nope its there for future.

File: src/ResourceManager/Websites/Commands.Websites/Properties/AssemblyInfo.cs (hunk 17 above)
Comment (NaveenGoli) [inline src/ResourceManager/Websites/Commands.Websites/Properties/AssemblyInfo.cs]: fixed

File: src/ResourceManager/Websites/Commands.Websites/Utilities/WebsitesClient.cs (hunk 19 above)
Comment (NaveenGoli) [inline src/ResourceManager/Websites/Commands.Websites/Utilities/WebsitesClient.cs]: removed..

File: src/ResourceManager/Websites/Commands.Websites/Properties/Resources.resx (hunk 18 above)
Comment (amitapl) [inline src/ResourceManager/Websites/Commands.Websites/Properties/Resources.resx]: So it should be at least empty

Comment (NaveenGoli) [inline src/ResourceManager/Websites/Commands.Websites/Properties/Resources.resx]: ok fixed it

File: src/ResourceManager/Websites/Commands.Websites/Cmdlets/NewAzureWebsites.cs (hunk 13 above)
Comment (NaveenGoli) [inline src/ResourceManager/Websites/Commands.Websites/Cmdlets/NewAzureWebsites.cs]: removed

**Comment by amitapl** [comment]
looks good

**Comment by NaveenGoli** [comment]
Will this be merged? AFAI can see, Microsoft.Azure.Commands.ScenarioTest.SqlTests.SecurityTests has no relation to websites code changes.. please let me know what the next steps are..

**Comment by matt-gibbs** [comment]
Not until the tests are passing. The reason New-AzureResourceGroup is not being found will need to be investigated.

**Comment by matt-gibbs** [comment]
re-test this please.

**Comment by NaveenGoli** [comment]
its passing on my local machine..

Assemblies Run 

C:\GitHUb\azure-powershell\src\ResourceManager\Sql\Commands.Sql.Test\bin\Debug\Microsoft.Azure.Commands.Sql.Test.dll

Summary 

Tests run: 3   Skipped: 2, Run time: 57.330s


Skipped tests 

Skipped 
❢
 Microsoft.Azure.Commands.ScenarioTest.SqlTests.SecurityTests.TestDatabaseUpdatePolicyWithStorage

Skipped 
❢
 Microsoft.Azure.Commands.ScenarioTest.SqlTests.SecurityTests.TestFailedDatabaseUpdatePolicyWithNoStorage


 All tests 

Click test class name to expand/collapse test details

57.070s  
✔
 Microsoft.Azure.Commands.ScenarioTest.SqlTests.SecurityTests   (3 tests)

**Comment by NaveenGoli** [comment]
when will this pull be taken in? I was under the impression it will happen today?

**Comment by yadavbdev** [comment]
There is one issue with this merge, the project does not build now as it is using the deprecated property of AzurePSCmdlet 'CurrentContext', where it should have been Profile.Context?

**Comment by NaveenGoli** [comment]
+Mark

Sent from my Windows Phone
________________________________
